model_templates/training/r_lang/custom.R

```diff
@@ -0,0 +1,158 @@
+init <- function(code_dir) {
+  # custom init function to load required libraries
+  library(tidyverse)
+  library(caret)
+  library(recipes)
+  library(e1071)
+  library(gbm)
+  source(paste(code_dir, '/create_pipeline.R', sep=''))
+}
+
+fit <- function(X, y, output_dir, class_order=NULL, row_weights=NULL, ...){
+    #' User-provided fit method, required for custom training
+    #'
+    #' Trains a regression or classification model using gbm (via caret)
+    #' @param X data.frame - training data to perform fit on
+    #' @param y data.frame column or array - target data to perform fit on
+    #' @param output_dir the path to write output. This is the path provided in '--output' parameter of the
+    #' 'drum fit' command.
+    #' @param class_order : A two element long list dictating the order of classes which should be used for
+    #'  modeling. Class order will always be passed to fit by DataRobot for classification tasks,
+    #'  and never otherwise. When models predict, they output a likelihood of one class, with a
+    #'  value from 0 to 1. The likelihood of the other class is 1 - this likelihood. Class order
+    #'  dictates that the first element in the list will be the 0 class, and the second will be the
+    #'  1 class.
+    #' @param row_weights: An array of non-negative numeric values which can be used to dictate how important
+    #'  a row is. Row weights is only optionally used, and there will be no filtering for which
+    #'  custom models support this. There are two situations when values will be passed into
+    #'  row_weights, during smart downsampling and when weights are explicitly provided by the user
+    #' @param ...: Added for forwards compatibility
+    #' @return Nothing
+
+  model <- create_pipeline(X, y)
+  # Save model
+  model_path <- file.path(
+    paste(
+      output_dir, 'artifact.rds', sep=''
+    )
+  )
+  saveRDS(model, file = model_path)
+}
+
+
+#"
+#    Below are all the hooks you can use to provide your own implementation.
+#    All hooks are currently commented out so uncomment a hook function in
+#    order to use it.
+#"
+#
+##' This hook can be implemented to adjust logic in the training and scoring mode.
+##'
+##' This hook is executed once the code is started.
+##' Load any libraries your serialized model requires so that scoring will work.
+##' For example, if your model was trained using method='brnn', you will need
+##' to add library(brnn) in this method
+##'
+##' @param ... keyword args, additional params to the method.
+##' code_dir - code folder passed in --code_dir argument
+##'
+##' @return NA, this method is intended for side effects only
+##' @export
+##'
+##' @examples
+#init <- function(...) {
+#}
+#
+##' This hook can be implemented to adjust logic in the scoring mode.
```

Comment:
I suggest either :
- remove inference hooks in order not to confuse potential users
or
- comment that those hooks are INFERENCE...

but removing is better.